### PR TITLE
Added permission checks

### PR DESCRIPTION
**Pull request by PizzaWhisperer**
Helps #56

### CODE REVIEW

**Comment by PizzaWhisperer** [comment]
Linked to #56

File: fs/fs.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ func HomeFolder() string {
 	return u.HomeDir
 }
 
+// CreateHomeConfigFolder creates a folder into home
```
Comment (nikkolasg) [inline fs/fs.go]: Feel free to remove since it is not used, as discussed.

File: fs/fs.go [hunk 2]
```diff
@@ -30,12 +31,22 @@ func CreateHomeConfigFolder(folder string) string {
...
 		if err := os.MkdirAll(folder, 0740); err != nil {
 			fmt.Println("folder", folder, ",err", err)
 			panic(err)
 		}
+	} else {
+		//the folder exists already
+		info, err := os.Lstat(folder)
+		perm := int(info.Mode().Perm())
+		//given that the permission is incremental (owner then group then others), then can compare the value of the permission bytes
+		if err != nil || perm < int(0740) {
```
Comment (nikkolasg) [inline fs/fs.go]: I am not sure I understand the reasoning. If the folder has perm `0777` then it is still considered as a "secure folder" ?
I think we should aim for an equality here, i.e. ` perm == 0740` no ?

File: fs/fs.go [hunk 3]
```diff
@@ -17,25 +17,20 @@ func HomeFolder() string {
...
-	}
-	path := path.Join(u.HomeDir, folder)
-	if exists, _ := Exists(path); !exists {
-		if err := os.MkdirAll(path, 0740); err != nil {
-			panic(err)
-		}
-	}
-	return path
-}
+// CreateSecureFolder checks if the folder exists and has the appropriate permission rights. If not it creates it.
```
Comment (nikkolasg) [inline fs/fs.go]: PLease add a comment saying it returns an empty string if there is an error with the permissions of an already existing folder.

**Pull request merged**